proposals/XXXX-stride.md

```diff
@@ -0,0 +1,208 @@
+
+This proposal extends `Sequence` to introduce strides across any base. 
+
+### Infinite and Non-numerical Sequences
+
+Swift's built-in `stride` functions are number based. They are useful for striding through integer and floating point numbers. They can act as integer-based collection indices. These functions aren't, however, suitable for skipping through non-numerical, possibly infinite sequences, which may or may not be multipass or otherwise indexable. 
+
+* For example, you might create a color progression `Sequence` and parameterize how quickly to procede from, for example, red through orange, yellow, green, and beyond.
+
+* You cannot use indexing features without creating an intermediate array.
+
+### Collections
+
+As `Collection` conforms to `Sequence`, this design adds strided array traversal. Therefore, you can stride through array members:
+
+```swift
+for value in myArray.striding(by: 5) {
+    // process every 5th member, starting with the
+    // first member of `myArray`:
+    //
+    // myArray[0], myArray[0 + strideLength], 
+    // myArray[0 + 2 * strideLength], ...
+    //
+}
+```
+
+This proposal makes no provision for offsetting the start or end points of a strided array but you can easily stride an array slice:
+
+```swift
+let results = Array(myArray[5...20].striding(by: 5))
+```
+
+As collections, you can stride both sets and dictionaries. The internal implementation details of sets and dictionaries will return the same strided results each time, although these collections are theoretically unordered. The documentation for `striding(by:)` does not call this out or provide any guarantees of repeatability.
+
+## Detailed Design
+
+```swift
+/// A strided non-contiguous sequence of elements that incorporates
+/// every nth element of a base sequence.
+///
+/// When used with collections, the sequence includes the first element
+/// followed by the element at the `startIndex` offset by the `stride`,
+/// by 2 * the `stride`, etc.
+///
+/// ```
+/// for value in myArray.striding(by: 5) {
+///   // process every 5th member, starting with the
+///   // first member of `myArray`:
+///   //
+///   // myArray[0], myArray[0 + strideLength],
+///   // myArray[0 + 2 * strideLength], ...
+/// }
+/// ```
+///
+/// To stride across a subsequence, use a collection slice:
+/// ```
+/// let results = Array(myArray[5...20].striding(by: 2))
+/// ```
+///
+public struct StridedSequence<BaseSequence: Sequence> : Sequence, IteratorProtocol {
+  public typealias Stride = Int
+  
+  public mutating func next() -> BaseSequence.Element? {
+    defer {
+      for _ in 0 ..< _strideLength - 1 {
+        let _ = _iterator.next()
```

Comment:
I recommend that you add some rationale for the trapping behavior; there is no reason that the `Sequence` implementation has to trap.